worlds/cvcotm/__init__.py

```diff
@@ -0,0 +1,228 @@
+                            f"({self.options.required_last_keys.value}) is higher than the Available Last Keys "
+                            f"({self.options.available_last_keys.value}). Lowering the required number to: "
+                            f"{self.required_last_keys}")
+            self.options.required_last_keys.value = self.required_last_keys
+
+    def create_regions(self) -> None:
+        # Create every Region object.
+        created_regions = [Region(name, self.player, self.multiworld) for name in get_all_region_names()]
+
+        # Attach the Regions to the Multiworld.
+        self.multiworld.regions.extend(created_regions)
+
+        for reg in created_regions:
+
+            # Add the Entrances to all the Regions.
+            ent_destinations_and_names = get_region_info(reg.name, "entrances")
+            if ent_destinations_and_names is not None:
+                reg.add_exits(ent_destinations_and_names)
+
+            # Add the Locations to all the Regions.
+            loc_names = get_region_info(reg.name, "locations")
+            if loc_names is None:
+                continue
+            locations_with_ids, locked_pairs = get_named_locations_data(loc_names, self.options)
+            reg.add_locations(locations_with_ids, CVCotMLocation)
+
+            # Place locked Items on all of their associated Locations.
+            for locked_loc, locked_item in locked_pairs.items():
+                self.get_location(locked_loc).place_locked_item(self.create_item(locked_item,
+                                                                                 ItemClassification.progression))
+
+    def create_item(self, name: str, force_classification: typing.Optional[ItemClassification] = None) -> Item:
+        if force_classification is not None:
+            classification = force_classification
+        else:
+            classification = cvcotm_item_info[name].default_classification
+
+        code = cvcotm_item_info[name].code
+        if code is not None:
+            code += BASE_ID
+
+        created_item = CVCotMItem(name, classification, code, self.player)
+
+        return created_item
+
+    def create_items(self) -> None:
+        item_counts = get_item_counts(self)
+
+        # Set up the items correctly
+        self.multiworld.itempool += [self.create_item(item, classification) for classification in item_counts for item
+                                     in item_counts[classification] for _ in range(item_counts[classification][item])]
+
+    def set_rules(self) -> None:
+        # Set all the Entrance and Location rules properly.
+        CVCotMRules(self).set_cvcotm_rules()
+
+    def pre_fill(self) -> None:
+        # Place the Double in local_early_items if the Early Double setting is on.
+        if self.options.early_double:
+            self.multiworld.local_early_items[self.player][iname.double] = 1
+
```

Comment:
Because of how early items are handled, this should be done in `generate_early` instead